### ISSUE TITLE
Discussion: Removing "react-native" from repo names and moving packages to @react-native-community

### ISSUE THREAD

**Issue by xzilja** [react-native-community/.github#33]
## Proposal

I want to propose 2 changes, that (in my opinion) could further improve "consistency" feeling around react-native-community org

1. Strip `react-native` from package names under `react-native-community` since organisation name already implies it.
2. Bring all packages under `@react-native-community/` namespace in package registry

## Discussion

1. How would maintainers feel about this?
2. Difficulties that can be caused by renaming repos in GitHub and possible solutions to solve them?
3. Difficulties that can be caused by renaming / moving packages into `@react-native-community/` namespace and possible solutions to solve them?
4. And in general is this something worth doing due to effort it requires (do all maintainers need to do this or can one person make the change across the org?)

## Present status

There are several examples that show case this proposal within the community already: [cli](https://github.com/react-native-community/cli), [apple-sign-in](https://github.com/react-native-community/apple-sign-in), [art](https://github.com/react-native-community/art) etc..

I also mentioned this in [react-native-svg](https://github.com/react-native-community/react-native-svg) repository via https://github.com/react-native-community/react-native-svg/issues/1021

**Comment by kelset** [comment]
Uhm I'm fairly sure there was already a conversation on this somewhere... I can't find it right now.

**Comment by matt-oakes** [comment]
Renaming Github repos is a good idea and shouldn't come with many downsides. Github automatically sets up redirects for the old names, so it should be fairly straightforward.

Releasing everything under the `@react-native-community` NPM scope would also be great, however, it might be a pain for users to need to migrate over and possibly leave some users who are unaware of the change "stranded" on older versions. Possible solutions:

* Release under the old NPM module and the new scoped one at the same time. Would be easier to manage with something like Semantic Release to do it for you.
* Mark the old module as "deprecated" to give users a warning that they need to migrate over: Details: https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions

Maybe it's worth talking to the maintainers of a library which has made a change to NPM package names about their experience with it? Babel comes to mind.

**Comment by pvinis** [comment]
Also, what do you mean by "bring all packages under RNC"?

**Comment by xzilja** [comment]
@pvins `@react-native-community` org over at npm

**Comment by thymikee** [comment]
Fully supportive, we already renamed a bunch, like CLI or ART.

**Comment by sibelius** [comment]
@sbeca wants to move https://github.com/react-native-community/react-native-audio-toolkit under `@react-native-community` scoped package

how can he do that?

**Comment by sbeca** [comment]
Yeah, [react-native-audio-toolkit](https://github.com/react-native-community/react-native-audio-toolkit) was transferred from Futurice's Github account to the react-native-community Github account a few months ago but is still under Futurice's npm account. What is the process for transferring the npm package to the react-native-community npm account?

**Comment by xzilja** [comment]
@thymikee when you have a minute, could you please outline steps that you guys took in order to make this happen? Once we have solid "miniguide" I could open proposal issues in each repo mentioning those (hopefully this won't be too spammy, but rather ensure all maintainers are aware?).

I guess we need to know how to: rename repo to remove 'react-native' and move npm package under rnc org with least resistance to maintainers and end users.

**Comment by thymikee** [comment]
1. Make sure the package is published under RNC scope to npm 
2. Go to the settings 
3. Change the repo name to be exactly the same as npm package 
4. Update references to the repo address if any (e.g links in error messages)

That's it, GH takes care of everything else for us.

If the package is not published as scoped, then it needs a discussion, but we should strive to have it sooner or later.

**Comment by matt-oakes** [comment]
If we're going to rename the NPM packages, we should have a plan to make it clear to existing users what they need to do to migrate. Possible options are:

* Continue to publish to both packages. This would be easier to do of using an automated tool.
* Mark the old package as deprecated and add information about the new name.
* Make sure the README has information about how to migrate, including how to redo the linking.

I recently went through this as a library user with RNC Blur View. I only noticed that the package name as changed after going to the GitHub page. If I hadn't have done that I would be "stuck" on the old version.

**Comment by sbeca** [comment]
So, for react-native-audio-toolkit I should do the following:
1. Publish v2.0.0 to NPM as a new NPM package: `@react-native-community/audio-toolkit`
2. Publish v1.0.7 to NPM under the old NPM package, which will just be the current version with a deprecation warning pointing to the new version.

I am not part of the react-native-community NPM organisation yet so who can help me with step 1? My NPM username is `scott.beca` if someone is looking to invite me to the org.

**Comment by MateusAndrade** [comment]
> I am not part of the react-native-community NPM organisation yet so who can help me with step 1? My NPM username is `scott.beca` if someone is looking to invite me to the org.

I think @kelset  can give you a hand with that 😃

**Comment by kelset** [comment]
👋 @sbeca.
Actually I don't have the NPM level of auth needed, but I'll ping who has.

In the meantime, there are a few other house cleaning things that I'd need your help with (for example, in the maintainers https://github.com/react-native-community/.github/blob/master/MAINTAINERS.md you are not listed 😅).

If you are on twitter, could you send me [a dm there](https://twitter.com/Kelset) so that I can guide you through a few of these things?

**Comment by sbeca** [comment]
I'll send you a DM now @kelset
